pinot-controller/src/main/java/com/linkedin/pinot/controller/api/resources/FileUploadPathProvider.java

```diff
@@ -66,22 +67,32 @@ public FileUploadPathProvider(ControllerConf controllerConf) throws InvalidContr
       Utils.rethrowException(e);
     }
 
+    String localTempDir = _controllerConf.getLocalTempDir();
+    if (localTempDir == null) {
+      LOGGER.warn("Local temp directory not set, setting to data dir {}", dataDir);
+      localTempDir = dataDir;
+    }
+
     PinotFS pinotFS = PinotFSFactory.create(scheme);
 
     try {
+      _localTempDirURI = new URI(localTempDir);
+      if (!pinotFS.exists(_localTempDirURI)) {
+        pinotFS.mkdir(_localTempDirURI);
+      }
       _baseDataDirURI = new URI(dataDir);
       if (!pinotFS.exists(_baseDataDirURI)) {
         pinotFS.mkdir(_baseDataDirURI);
       }
-      _fileUploadTmpDirURI = new URI(_baseDataDirURI + FILE_UPLOAD_TEMP_PATH);
+      _fileUploadTmpDirURI = new URI(_localTempDirURI + FILE_UPLOAD_TEMP_PATH);
```

Comment:
Do we need a '/' between tempdir and upload path?

Reply:
my paths include the slash